src/ts/background.js

```diff
@@ -1,19 +1,16 @@
 import jq from 'jq-web/jq.wasm.js';
 
-function getJq(jsonInput, filter = '.') {
-    return jq(JSON.parse(jsonInput), filter);
-}
-
 // Build the connection to the frontend
 chrome.runtime.onConnect.addListener(port => {
     // Check the connection
     if (port.name === 'jq-backend-connection') {
         // Listen to messages from frontend
         port.onMessage.addListener(msg => {
             // Filter the JSON with JQ
-            const jqResult = JSON.stringify(getJq(msg.json, msg.filter));
-            // Send the result from JQ to the frontend
-            port.postMessage({jqResult: jqResult});
+            jq.promised.raw(msg.json, msg.filter).then(result => {
```

Comment:
you have a variable mismatch. 
you use `result` on line 10 but then `jqResult` on like 12
please change line 10 to `jqResult`